packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -156,6 +156,102 @@ contract OptimismPortal_Test is CommonTest {
         assertEq(address(optimismPortal).balance, _value);
     }
 
+    /// @dev Tests that `depositTransaction` succeeds when msg.sender == tx.origin and non-custom gas is used.
+    function testFuzz_depositTransaction_senderIsOrigin_succeeds(
+        address _to,
+        uint256 _mint,
+        uint256 _value,
+        uint64 _gasLimit,
+        bool _isCreation,
+        bytes memory _data
+    )
+        external
+    {
+        if (_isCreation) {
+            vm.assume(_to == address(0));
+        }
+        vm.assume(_data.length <= 120_000);
+        vm.assume(_gasLimit >= optimismPortal.minimumGasLimit(uint64(_data.length)));
+        ResourceMetering.ResourceConfig memory rcfg = systemConfig.resourceConfig();
+        vm.assume(int256(uint256(_gasLimit)) <= int256(uint256(rcfg.maxResourceLimit)));
+
+        uint256 prevBalance = address(optimismPortal).balance;
+
+        // Ensure that no custom gas token is set
+        (address gasPayingToken,) = optimismPortal.gasPayingToken();
+        assertEq(gasPayingToken, Constants.ETHER);
+
+        bytes memory opaqueData = abi.encodePacked(_mint, _value, _gasLimit, _isCreation, _data);
+
+        vm.expectEmit(address(optimismPortal));
+        emit TransactionDeposited(
+            address(this), // from
+            _to,
+            uint256(0), // DEPOSIT_VERSION
+            opaqueData
+        );
+
+        vm.deal(address(this), _mint);
+
+        // Ensure that msg.sender == tx.origin
+        vm.prank(address(this), address(this));
+
+        // Deposit the token into the portal
+        optimismPortal.depositTransaction{ value: _mint }(_to, _value, _gasLimit, _isCreation, _data);
+
+        // Assert final balance equals the deposited amount
+        assertEq(address(optimismPortal).balance, _mint + prevBalance);
+        assertEq(optimismPortal.balance(), _mint + prevBalance);
+    }
+
+    /// @dev Tests that `depositTransaction` succeeds when msg.sender != tx.origin and non-custom gas is used.
+    function testFuzz_depositTransaction_senderNotOrigin_succeeds(
```

Comment:
You could reduce code duplication with an internal function that is called by both of the 2 external fuzz tests